src/Components/Auth/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,244 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { ScaleLoader } from "react-spinners";
+import { css } from "@emotion/core";
+import { toast } from "react-toastify";
+import { Link } from "react-router-dom";
+
+import Sidebar from "../../Shared/AuthSidebar/AuthSidebar.jsx";
+import "./SignUp.scss";
+import logo from "../../../Assets/logo.svg";
+import facebookLogo from "../../../Assets/facebookLogo.svg";
+import googleLogo from "../../../Assets/googleLogo.svg";
+import { signUp } from "../../../Redux/Actions/SignUp/signUpAction";
+
+const loaderCss = css`
+  display: inline-block;
+`;
+/**
+ * This class contains methods
+ * for signup component
+ */
+export class SignUp extends Component {
+  /**
+   * This method has a constructor which calls super() method
+   * and initialize state object
+   * @param {object} props props passed to this class.
+   * @returns {void}
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstname: "",
+      lastname: "",
+      username: "",
+      email: "",
+      password: ""
+    };
+  }
+
+  /**
+   * This method handles onChange event.
+   * @param {object} event event information.
+   * @returns {void}
+   */
+  onChange = event => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value
+    });
+  };
+
+  /**
+   * This method handles onSubmit event.
+   * @param {object} event event information.
+   * @returns {void}
+   */
+  onSubmit = async event => {
+    event.preventDefault();
+    const { signUpAction } = this.props;
+    const { firstname, lastname, username, email, password } = this.state;
+    const form = {
+      firstname,
+      lastname,
+      username,
+      email,
+      password
+    };
+    await signUpAction(form);
```

Comment:
Kindly remove `async `and `await ` as you have already used them in actions

Reply:
Done